### PR TITLE
PR: Update RELEASE.md and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* Some outdated references to spyder `4.x` and spyder-kernels `1.x` branches.
* Update crowdin link to sync translations
* Update `conda.txt` requirements link

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ To release a new version of Spyder you need to follow these steps:
 
 * In case a new version is not part of the `defaults` channel yet, you need to copy it to our channel with the following command:
```
Comment (ccordoba12) [inline RELEASE.md:94]: This is not necessary anymore because we switched to run our tests with Conda-forge. So please remove this bullet point.

**Review by ccordoba12 — COMMENTED** [review]
Thanks a lot for this @dalthviz! I left a minor comment, otherwise looks good to me.

**Review by ccordoba12 — APPROVED** [review]
Thanks @dalthviz!

**Comment by texadactyl** [comment]
I know much work it is to keep a repository's *.md files up to date and consistent with the rest of the repository so thank you.

Good job!

**Pull request merged**